### PR TITLE
Well init restart bug

### PR DESCRIPTION
**Pull request by joakim-hove**
Update: after a review comment from @bska this code has been completely reworked.

If some connections have been closed there will be a difference between the number of connections in the well model and the number of connections loaded from the restart file. This should take that into account.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 1]
```diff
@@ -1916,34 +1916,81 @@ namespace Opm {
...
+                        const auto& rst_connection = rst_well.connections[sched_connection_index];
+
+                        perf_pressure[local_connection_index] = rst_connection.pressure;
+                        perf_rates[local_connection_index] = rst_connection.reservoir_rate;
+                        for (int phase_index = 0; phase_index < np; ++phase_index)
+                            perf_phase_rates[local_connection_index*np + phase_index] = rst_connection.rates.get(phs[phase_index]);
+
+                        local_connection_index++;
+                    }
                 }
```
Comment (bska) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp:1982]: If the [`well_perf_data_`](https://github.com/OPM/opm-simulators/blob/061b5c2395e0928885c714c09dbea2447961dc54/opm/simulators/wells/BlackoilWellModel.hpp#L405) is set up by the time we get here, I think you can probably simplify the logic through [`PerforationData::ecl_index`](https://github.com/OPM/opm-simulators/blob/061b5c2395e0928885c714c09dbea2447961dc54/opm/simulators/wells/PerforationData.hpp#L32-L33)

Comment (joakim-hove) [reply opm/simulators/wells/BlackoilWellModel_impl.hpp]: Thanks - that was a good suggestion; completely rewritten.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
@totto82 : I have marked this with "Release 2021.04" as I think it is a quite important bug fix.

**Comment by totto82** [comment]
Agree, I will back-port it when it is merged.

**Comment by bska** [comment]
> I will back-port it when it is merged.

Feel free to do so at any time because I'm merging this now.

**Pull request merged**